addons/payment/models/ir_module_module.py

```diff
@@ -0,0 +1,55 @@
+# coding: utf-8
+
+from odoo import api, fields, models
+
+
+class IrModule(models.Model):
+    _name = 'ir.module.module'
+    _inherit = 'ir.module.module'
+
+    payment_environment = fields.Selection([
+        ('test', 'Test'),
+        ('prod', 'Production')], 'Acquirer Environment',
+        compute='_compute_payment_environment', inverse='_set_payment_environment',
+        search='_search_payment_environment')
+    payment_acquirer_id = fields.Many2one(
+        'payment.acquirer', compute='_compute_payment_acquirer_id')
+
+    @api.depends('state')
+    def _compute_payment_acquirer_id(self):
+        payment_modules = self.filtered(lambda module: module.name.startswith('payment_'))
+        payment_names = map(lambda name: name.split('payment_', 1)[1], payment_modules.mapped('name'))
+        payment_acquirers = dict((acquirer.provider, acquirer.id) for acquirer in self.env['payment.acquirer'].search([('provider', 'in', payment_names)]))
```

Comment:
dictcomp motherfucker, do you use it?
![1200](https://cloud.githubusercontent.com/assets/7571158/20310060/c65c9ce4-ab4a-11e6-8a49-f37c3f8e2887.jpg)